packages/hash/api/src/db/postgres/aggregation/sql/aggregations.util.ts

```diff
@@ -0,0 +1,234 @@
+      aggregationVersionsColumnNames.filter(
+        (column) => !aggregationsColumnNames.includes(column),
+      ),
+      "aggregation_versions",
+    )}
+  from
+    aggregations
+    join aggregation_versions
+    on aggregations.aggregation_id = aggregation_versions.aggregation_id
+`;
+
+/** selects all aggregations in the datastore with their latest version */
+const selectAllLatestVersionsOfAggregations = sql<DbAggregationWithVersionRow>`
+  select distinct on (aggregation_id) *
+  from (${selectAllAggregationVersions}) as all_aggregation_versions
+  order by aggregation_id, updated_at desc
+`;
+
+/** select all aggregations with the latest version before a given timestamp */
+const selectAllLatestVersionsOfAggregationsBeforeTimestamp = (params: {
+  beforeTimestamp: Date;
+}) =>
+  sql<DbAggregationWithVersionRow>`
+    select distinct on (aggregation_id) *
+    from (
+      ${selectAllAggregationVersions}
+      where aggregation_versions.updated_at <= ${params.beforeTimestamp.toISOString()}
+    ) as all_aggregation_versions
+    order by aggregation_id, updated_at desc
+`;
+
+/** selects all versions of a specifc aggregation */
+const selectAllVersionsOfAggregation = (params: {
+  sourceAccountId: string;
+  aggregationId: string;
+}) => sql<DbAggregationWithVersionRow>`
+  with all_aggregations as (${selectAllAggregationVersions})
+  select *
+  from all_aggregations
+  where
+    source_account_id = ${params.sourceAccountId} and aggregation_id = ${params.aggregationId}
+`;
+
+/** selects the latest version of a specific aggregation */
+export const selectLatestVersionOfAggregation = (params: {
+  sourceAccountId: string;
+  aggregationId: string;
+}) => sql<DbAggregationWithVersionRow>`
+  with all_aggregation_versions as (${selectAllVersionsOfAggregation(params)})
+  select distinct on (aggregation_id) *
+  from all_aggregation_versions
+  order by aggregation_id, updated_at desc
+`;
+
+export const selectAllAggregationsWithSourceEntity = (params: {
+  sourceAccountId: string;
+  sourceEntityId: string;
+  activeAt?: Date;
+  path?: string;
+  additionalClauses?: ValueExpressionType[];
+}) => sql<DbAggregationWithVersionRow>`
+ with all_aggregations as (${
+   params.activeAt
+     ? selectAllLatestVersionsOfAggregationsBeforeTimestamp({
+         beforeTimestamp: params.activeAt,
+       })
+     : selectAllLatestVersionsOfAggregations
+ })
+  select *
+  from all_aggregations
+  where
+    ${sql.join(
+      [
+        sql`source_account_id = ${params.sourceAccountId}`,
+        sql`source_entity_id = ${params.sourceEntityId}`,
+        params.activeAt
+          ? [
+              // the aggregation was applied before the timestamp
+              sql`applied_to_source_at <= ${params.activeAt.toISOString()}`,
+              // either the aggregation was removed after the timestamp, or the aggregation hasn't been removed yet
+              sql`(
+                  removed_from_source_at >= ${params.activeAt.toISOString()}
```

Comment:
If the aggregation was removed at the time I requested, I wouldn't consider it active at that time
```suggestion
                  removed_from_source_at > ${params.activeAt.toISOString()}
```

Reply:
Good point, I've addressed this and the same mistake made in the `selectAllLinksWithSourceEntity` query in https://github.com/hashintel/hash/pull/484/commits/601bfa7b76c072963ff9193d1f68e2028794dae8